### ISSUE TITLE
Feature request: make work on MSYS2 window-based systems

### ISSUE THREAD

**Issue by hzeller** [hendrikmuhs/ccache-action#24]
Thanks for this action, I am successfully using it in various projects already :)

There are some projects I am involved in that test compiling on windows using the [msys2](https://github.com/msys2/setup-msys2) action, which essentially installs a gcc etc on the winows host, more closely making it behave like a linux machine. But compiling on msys2 (at least the github action CI runners) in particular is [painfully slow](https://github.com/chipsalliance/UHDM/actions/runs/1225192359) so I wonder if your action could also be made work there.

I did try, but ... I am not familiar with typescript nor msys, so my simple attempt (see below) failed.

It does look like that maybe paths in msys2 need to be set up properly as this one failed to [pacman install](https://github.com/hzeller/UHDM/runs/3576210481?check_suite_focus=true) the package.

But even if  I [manually install](https://github.com/hzeller/UHDM/blob/ff434a2f45e7e443392bddc06bc7f093ef5dd067/.github/workflows/main.yml#L95) `ccache` in msys, and then use something essentially like [`export CXX="ccache g++"`](https://github.com/hzeller/UHDM/blob/ff434a2f45e7e443392bddc06bc7f093ef5dd067/.github/workflows/main.yml#L116-L117), just invocating `$CXX` hangs, so the makefile does not go anywyere. So there is a compound problem going on that someone more familiar with Windows  and the interactions there might be more easily be able to troubleshoot.

```diff
diff --git a/dist/restore/index.js b/dist/restore/index.js
index 87b50ab..10db824 100644
--- a/dist/restore/index.js
+++ b/dist/restore/index.js
@@ -59239,8 +59239,9 @@ var cache = __nccwpck_require__(7799);
 async function install() {
     if (external_process_namespaceObject.platform === "darwin") {
         await exec.exec("brew install ccache");
-    }
-    else {
+    } else if (process.platform === "win32") {
+        await exec.exec("pacman --noconfirm -S --needed ccache");
+    } else {
         await exec.exec("sudo apt-get install -y ccache");
     }
 }
@@ -59298,4 +59299,4 @@ run();
 
 module.exports = __webpack_exports__;
 /******/ })()
-;
\ No newline at end of file
+;
diff --git a/src/restore.ts b/src/restore.ts
index d7e49d7..d4b8f33 100644
--- a/src/restore.ts
+++ b/src/restore.ts
@@ -9,6 +9,8 @@ import * as cache from "@actions/cache";
 async function install() {
   if (process.platform === "darwin") {
     await exec.exec("brew install ccache");
+  } else if (process.platform === "win32") {
+     await exec.exec("pacman --noconfirm -S --needed ccache");
   } else {
     await exec.exec("sudo apt-get install -y ccache");
   }
```

**Comment by hendrikmuhs** [comment]
Thanks for the feedback. As stated in the other issue, I don't have a windows build environment I could use for implementation/testing.

However, I happily help and merge once its working.

> I did try, but ... I am not familiar with typescript nor msys, so my simple attempt (see below) failed.

After changing typescript, you need to compile it into javascript and upload the changed files which are created in `dist/`. 

For this you need `npm`, the command to execute is `npm install`.

**Comment by MaxPeal** [comment]
i haven't tried it but, on windows maybe take a look on this other ccache GitHub action
https://cristianadam.eu/20200113/speeding-up-c-plus-plus-github-actions-using-ccache/

**Comment by jonashaag** [comment]
See also https://github.com/hendrikmuhs/ccache-action/pull/47

**Comment by jonashaag** [comment]
#50

**Comment by hendrikmuhs** [comment]
fixed in #50
